avm/ptn/authorization/resource-role-assignment/main.bicep

```diff
@@ -0,0 +1,106 @@
+metadata name = 'ResourceRole Assignments (All scopes)'
+metadata description = 'This module deploys a Role Assignment for a specific resource.'
+metadata owner = 'Azure/module-maintainers'
+
+@sys.description('Required. The scope for the role assignment, fully qualified resourceId.')
+param resourceId string
+
+@sys.description('Required. The unique guid name for the role assignment.')
+param name string
+
+@sys.description('Required. The role definition ID for the role assignment.')
+param roleDefinitionId string
+
+@sys.description('Optional. The name for the role, used for logging.')
+param roleName string = ''
+
+@sys.description('Required. The Principal or Object ID of the Security Principal (User, Group, Service Principal, Managed Identity).')
+param principalId string
+
+@sys.description('Optional. The principal type of the assigned principal ID.')
+@allowed([
+  'ServicePrincipal'
+  'Group'
+  'User'
+  'ForeignGroup'
+  'Device'
+  ''
+])
+param principalType string = ''
+
+@sys.description('Optional. The description of role assignment.')
+param description string = ''
+
+@sys.description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.authorization-resourceroleassignment.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+// Workaround to make Pester test work.
+// Simple loadJsonContent creates a variable in main.json
```

Comment:
Hey @peterbud would you happen to a have a link to a failed workflow run available? I'd like to investigate this further. Naturally, we don't need to block the PR over it, but it sounds like something we should fix in the tests...

Reply:
Hi @AlexanderSehr , here is the [failed workflow](https://github.com/peterbud/bicep-registry-modules/actions/runs/9024003041/job/24797001028#step:4:190)

Also I have noticed one similar case, when `loadJsonContent' creates a simelar non camel case variable here at the [storage-account/file-service](https://github.com/Azure/bicep-registry-modules/blob/dcd987fd8211f5dcd762eccd3c9ef3791e104838/avm/res/storage/storage-account/file-service/share/main.json#L227)

Reply:
Much appreciated, thanks @peterbud. I'll add it to my todo list. 
And ah right, thanks for reminding me of this horrible `fileShares` workaround I added some time ago. Almost successfully suppressed it from my memory 😄 Interesting though, that the static tests seemed not to have failed for the same/similar variable there though.

Anyways, will resolve this comment and make a note to investigate.